.github/workflows/js-checks.yml

```diff
@@ -16,7 +16,7 @@ jobs:
   run-checks:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
```

Comment:
This will prevent us from getting updates done to this action, even bugfixes. `actions/checkout` and others are well known for dealing with semver the right way so I think sticking to i.e. `v4` in this case is safe and requires less future maintenance.

Reply:
That same doc also says "Pin actions to a tag only if you trust the creator" and the author of `actions/checkout` is GitHub itself. Additionally, how do you know the version pinned (v4.2.1) is not already compromised?

Reply:
`actions/checkout` has commits from 57 accounts: https://github.com/actions/checkout/graphs/contributors - do we really trust all 57 of these accounts?

I don't trust that it isn't already compromised, especially given the bundle contains 38k lines of code: https://github.com/actions/checkout/blob/main/dist/index.js

Reply:
So, at the end of the day we do not trust `eef614` neither we trust `v4`. And that will not change unless we audit their code. In summary:

Fixing the version:
  Pros: If a new vulnerability is introduced we will not be affected.
  Cons: If there is any current vulnerability, we will not get patches unless we find it and change the hash.
Using a tag:
  Pros: If the community finds any current or future vulnerability, patches will come automatically.
  Cons: If a current or future vulnerability is not patched, we (and everyone) could be attacked.
  
I lean towards using a tag -for this specific case- as the open-source development model has proven to be quite effective in dealing with security issues. From my PoV, convenience+flexibility compensate the risks.

Reply:
hey everyone, sorry I am late to review this PR, I have been thinking about tag vs hash and have some thoughts 🤔 

I prefer hashes because I like to know exactly what I am working with.  Aside from security, having a hash would ensure that you're not debugging through something that doesn't match what someone else found an issues with (with either their local setup or something that is deployed).

Also, I don't think we should strategize on having stochastic security patches (that we're unaware of) via people pushing to the tag that we have deployed.  I think that the cases where this would _benefit_ us would be few.

So, I would vote for hashes for two reasons:
* functionality predictability
* not relying on stochastic updates as a security strategy

Reply:
I'd go so far at to say that unless a tag is a permanent identifier that cannot change (which sadly, git tags are not) I do not want tags and prefer hashes.  Any changes to a build system that we do not initiate are a potential issue in my opinion.

Reply:
There are almost 90 references to `actions/checkout@v*` across all `hemilabs` repos and many more to other actions. What are you guys planning to do with those?

Reply:
Ideally every repository would use actions from this repository, then they are all updated in once place.

Reply:
I am sorry, but I do not understand why we are arguing about this.
1. Pinning all actions is a very common best practice, even more so for reusable workflows since they are updated in one place.
2. Pinning actions improves overall security, especially here where supply-chain security is critical.
3. Pinning actions helps with stability, as actions will not change unless we change them.
4. More people seem to be for pinning vs using tags.

I think the benefits of pinning are so much higher than not, and I am happy to be the one responsible for keeping these actions up-to-date.

Reply:
Because in theory those arguments are rock solid. But reality is much more complex. And this discussion needs to end up with a policy to define how we do this at Hemi.

Given the sources of the action were not audited, trusting any action version is equivalent. That applies to current or future actions. Fixing the version ensures no security patches or bug fixes will be automatically incorporated, widening the possibility of attacks.

And in addition, being targeted through these very well known actions like actions/checkout that are used in every single repository is IMO highly unlikely. It is more likely that we are affected by a bug because of using an old version instead. This last problem hit us a few times. Not the former.

But of course we should use hashes for other actions we chose not to trust *and* audit to ensure we are not compromised *already*.

For these well known actions, using just the major version (i.e. `@v4`) unintuitively minimizes the risks while provides flexibility and low/no maintenance effort.

Reply:
Every best practice I've seen says to use hashes.  We clearly don't do that everywhere but we should.  The point of this repo is to standardize what we do.  We aren't fully there yet, but doing it here is a start.

I agree that I would love to only use audited actions, but we lack the manpower to do that in any reasonable way.  So at least using a known version rather than one that can change is the best we can do.